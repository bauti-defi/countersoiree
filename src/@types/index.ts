import {ethers} from 'ethers';

export type EtherscanNetwork =
  | 'mainnet'
  | 'morden'
  | 'ropsten'
  | 'rinkeby'
  | 'arbitrum'
  | 'arbitrum_rinkeby'

export type EtherscanProps = {
  readonly etherscanKey: string;
  readonly network: EtherscanNetwork;
};

export type PendingTransaction = ethers.providers.TransactionResponse;
