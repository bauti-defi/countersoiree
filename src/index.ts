import 'graphql-import-node';

export * from './@types';
export * from './ethereum';
export * from './graphql';
