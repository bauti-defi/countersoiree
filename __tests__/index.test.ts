import 'jest';

import * as fs from 'fs';
import * as path from 'path';

import {ethers} from 'ethers';

import {getMaybeFunctionForMaybeInterface, maybeInterfaceToFunctionIdentifier} from '../src';

const OPENSEA_SEAPORT_V_1_1_FULFILL_BASIC_ORDER = /* ethers */
  'function fulfillBasicOrder(tuple(address,uint256,uint256,address,address,address,uint256,uint256,uint8,uint256,uint256,bytes32,uint256,bytes32,bytes32,uint256,tuple(uint256,address)[],bytes)) payable returns (bool)';

function loadFixture<T = unknown>({path: p}: {
  readonly path: string;
}) {
  return JSON.parse(fs.readFileSync(path.resolve('__fixtures__', `${p}.json`), 'utf-8')) as T;
}

const loadSeaportv1_1 = () => new ethers.utils.Interface(
  loadFixture({path: 'Seaport_v1_1'}),
);

describe('countersoiree', () => {
  it('seaport_v11::abi', () => {
    expect(loadSeaportv1_1()).toBeTruthy();

    [undefined, '', null]
      .forEach(maybeInterface => expect(
        getMaybeFunctionForMaybeInterface({
          abi: loadSeaportv1_1(),
          maybeInterface,
        }),
      ).toBe(false),
    );

    expect(
      maybeInterfaceToFunctionIdentifier({
        interfaceString: OPENSEA_SEAPORT_V_1_1_FULFILL_BASIC_ORDER,
      })
    ).toBe('fulfillBasicOrder');

    expect(
      getMaybeFunctionForMaybeInterface({
        abi: loadSeaportv1_1(),
        maybeInterface: OPENSEA_SEAPORT_V_1_1_FULFILL_BASIC_ORDER,
      }),
    ).toMatchSnapshot();

  });
});
