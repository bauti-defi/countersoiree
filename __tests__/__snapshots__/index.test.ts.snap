// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`countersoiree seaport_v11::abi 1`] = `
FunctionFragment {
  "_isFragment": true,
  "constant": false,
  "gas": null,
  "inputs": [
    ParamType {
      "_isParamType": true,
      "arrayChildren": null,
      "arrayLength": null,
      "baseType": "tuple",
      "components": [
        ParamType {
          "_isParamType": true,
          "arrayChildren": null,
          "arrayLength": null,
          "baseType": "address",
          "components": null,
          "indexed": null,
          "name": "considerationToken",
          "type": "address",
        },
        ParamType {
          "_isParamType": true,
          "arrayChildren": null,
          "arrayLength": null,
          "baseType": "uint256",
          "components": null,
          "indexed": null,
          "name": "considerationIdentifier",
          "type": "uint256",
        },
        ParamType {
          "_isParamType": true,
          "arrayChildren": null,
          "arrayLength": null,
          "baseType": "uint256",
          "components": null,
          "indexed": null,
          "name": "considerationAmount",
          "type": "uint256",
        },
        ParamType {
          "_isParamType": true,
          "arrayChildren": null,
          "arrayLength": null,
          "baseType": "address",
          "components": null,
          "indexed": null,
          "name": "offerer",
          "type": "address",
        },
        ParamType {
          "_isParamType": true,
          "arrayChildren": null,
          "arrayLength": null,
          "baseType": "address",
          "components": null,
          "indexed": null,
          "name": "zone",
          "type": "address",
        },
        ParamType {
          "_isParamType": true,
          "arrayChildren": null,
          "arrayLength": null,
          "baseType": "address",
          "components": null,
          "indexed": null,
          "name": "offerToken",
          "type": "address",
        },
        ParamType {
          "_isParamType": true,
          "arrayChildren": null,
          "arrayLength": null,
          "baseType": "uint256",
          "components": null,
          "indexed": null,
          "name": "offerIdentifier",
          "type": "uint256",
        },
        ParamType {
          "_isParamType": true,
          "arrayChildren": null,
          "arrayLength": null,
          "baseType": "uint256",
          "components": null,
          "indexed": null,
          "name": "offerAmount",
          "type": "uint256",
        },
        ParamType {
          "_isParamType": true,
          "arrayChildren": null,
          "arrayLength": null,
          "baseType": "uint8",
          "components": null,
          "indexed": null,
          "name": "basicOrderType",
          "type": "uint8",
        },
        ParamType {
          "_isParamType": true,
          "arrayChildren": null,
          "arrayLength": null,
          "baseType": "uint256",
          "components": null,
          "indexed": null,
          "name": "startTime",
          "type": "uint256",
        },
        ParamType {
          "_isParamType": true,
          "arrayChildren": null,
          "arrayLength": null,
          "baseType": "uint256",
          "components": null,
          "indexed": null,
          "name": "endTime",
          "type": "uint256",
        },
        ParamType {
          "_isParamType": true,
          "arrayChildren": null,
          "arrayLength": null,
          "baseType": "bytes32",
          "components": null,
          "indexed": null,
          "name": "zoneHash",
          "type": "bytes32",
        },
        ParamType {
          "_isParamType": true,
          "arrayChildren": null,
          "arrayLength": null,
          "baseType": "uint256",
          "components": null,
          "indexed": null,
          "name": "salt",
          "type": "uint256",
        },
        ParamType {
          "_isParamType": true,
          "arrayChildren": null,
          "arrayLength": null,
          "baseType": "bytes32",
          "components": null,
          "indexed": null,
          "name": "offererConduitKey",
          "type": "bytes32",
        },
        ParamType {
          "_isParamType": true,
          "arrayChildren": null,
          "arrayLength": null,
          "baseType": "bytes32",
          "components": null,
          "indexed": null,
          "name": "fulfillerConduitKey",
          "type": "bytes32",
        },
        ParamType {
          "_isParamType": true,
          "arrayChildren": null,
          "arrayLength": null,
          "baseType": "uint256",
          "components": null,
          "indexed": null,
          "name": "totalOriginalAdditionalRecipients",
          "type": "uint256",
        },
        ParamType {
          "_isParamType": true,
          "arrayChildren": ParamType {
            "_isParamType": true,
            "arrayChildren": null,
            "arrayLength": null,
            "baseType": "tuple",
            "components": [
              ParamType {
                "_isParamType": true,
                "arrayChildren": null,
                "arrayLength": null,
                "baseType": "uint256",
                "components": null,
                "indexed": null,
                "name": "amount",
                "type": "uint256",
              },
              ParamType {
                "_isParamType": true,
                "arrayChildren": null,
                "arrayLength": null,
                "baseType": "address",
                "components": null,
                "indexed": null,
                "name": "recipient",
                "type": "address",
              },
            ],
            "indexed": null,
            "name": null,
            "type": "tuple",
          },
          "arrayLength": -1,
          "baseType": "array",
          "components": [
            ParamType {
              "_isParamType": true,
              "arrayChildren": null,
              "arrayLength": null,
              "baseType": "uint256",
              "components": null,
              "indexed": null,
              "name": "amount",
              "type": "uint256",
            },
            ParamType {
              "_isParamType": true,
              "arrayChildren": null,
              "arrayLength": null,
              "baseType": "address",
              "components": null,
              "indexed": null,
              "name": "recipient",
              "type": "address",
            },
          ],
          "indexed": null,
          "name": "additionalRecipients",
          "type": "tuple[]",
        },
        ParamType {
          "_isParamType": true,
          "arrayChildren": null,
          "arrayLength": null,
          "baseType": "bytes",
          "components": null,
          "indexed": null,
          "name": "signature",
          "type": "bytes",
        },
      ],
      "indexed": null,
      "name": "parameters",
      "type": "tuple",
    },
  ],
  "name": "fulfillBasicOrder",
  "outputs": [
    ParamType {
      "_isParamType": true,
      "arrayChildren": null,
      "arrayLength": null,
      "baseType": "bool",
      "components": null,
      "indexed": null,
      "name": "fulfilled",
      "type": "bool",
    },
  ],
  "payable": true,
  "stateMutability": "payable",
  "type": "function",
}
`;
